integration/mesos/src/test/java/alluxio/mesos/AlluxioFrameworkIntegrationTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.mesos;
+
+import static org.junit.Assert.assertEquals;
+
+import alluxio.AlluxioURI;
+import alluxio.CommonTestUtils;
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.client.ClientContext;
+import alluxio.client.block.BlockMasterClient;
+import alluxio.client.block.RetryHandlingBlockMasterClient;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.exception.ConnectionFailedException;
+import alluxio.util.CommonUtils;
+import alluxio.util.io.PathUtils;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.google.common.base.Function;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.io.IOUtils;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Integration tests for {@link AlluxioFramework}. These tests assume that a Mesos cluster is
+ * running locally.
+ */
+public class AlluxioFrameworkIntegrationTest {
```

Comment:
It's purpose is definitely to test Mesos, not to serve as an example of how we want people to use Mesos. It's similar to JournalCrashTest though, so I'm not sure - do you think it makes more sense in examples/?